tests/index/tests.c

```diff
@@ -253,6 +253,56 @@ void test_index_tests__add(void)
 	git_repository_free(repo);
 }
 
+void test_index_tests__add_frombuffer(void)
+{
+	git_index *index;
+	git_repository *repo;
+	const git_index_entry *entry;
+	git_oid id1;
+	git_blob *blob;
+
+	cl_set_cleanup(&cleanup_myrepo, NULL);
+
+	/* Intialize a new repository */
+	cl_git_pass(git_repository_init(&repo, "./myrepo", 0));
+
+	/* Ensure we're the only guy in the room */
+	cl_git_pass(git_repository_index(&index, repo));
+	cl_assert(git_index_entrycount(index) == 0);
+
+	/* Store the expected hash of the file/blob
+	 * This has been generated by executing the following
+	 * $ echo "hey there" | git hash-object --stdin
+	 */
+	cl_git_pass(git_oid_fromstr(&id1, "a8233120f6ad708f843d861ce2b7228ec4e3dec6"));
+
+	/* Add the new file to the index */
+	cl_git_pass(git_index_add_frombuffer(index, "test.txt", GIT_FILEMODE_BLOB,
+		"hey there\n", 10));
+
+	/* Wow... it worked! */
+	cl_assert(git_index_entrycount(index) == 1);
+	entry = git_index_get_byindex(index, 0);
+
+	/* And the built-in hashing mechanism worked as expected */
+	cl_assert_equal_oid(&id1, &entry->id);
+	/* And mode is the one asked */
+	cl_assert(entry->mode == GIT_FILEMODE_BLOB);
```

Comment:
Let's use `cl_assert_equal_i(GIT_FILEMODE_BLOB, entry->mode)`. This way clar will report the values.